python/ray/data/_internal/util.py

```diff
@@ -453,6 +453,9 @@ def pandas_df_to_arrow_block(df: "pandas.DataFrame") -> "Block":
 def ndarray_to_block(ndarray: np.ndarray, strict_mode: bool) -> "Block":
     from ray.data.block import BlockAccessor, BlockExecStats
 
+    ctx = ray.data.DataContext.get_current()
+    ctx.strict_mode = strict_mode
```

Comment:
This will change the global strict_mode config. 
I guess "strict_mode" arg is supposed to only apply to this function?

Reply:
Let me change this to propagate the entire context properly.